src/crewai/agent.py

```diff
@@ -245,27 +257,65 @@ def execute_task(
         knowledge_config = (
             self.knowledge_config.model_dump() if self.knowledge_config else {}
         )
+
         if self.knowledge:
-            agent_knowledge_snippets = self.knowledge.query(
-                [task.prompt()], **knowledge_config
+            crewai_event_bus.emit(
+                self,
+                event=KnowledgeRetrievalStartedEvent(
+                    agent=self,
+                ),
             )
-            if agent_knowledge_snippets:
-                self.agent_knowledge_context = extract_knowledge_context(
-                    agent_knowledge_snippets
+            try:
+                self.knowledge_search_query = self._get_knowledge_search_query(
```

Comment:
What happens when the `self.knowledge_search_query` is being set? It seems we are overwriting it. 
If it's the expected behavior, shouldn't it be a PrivateAttr instead? It does make sense?

Reply:
Just double checking..
I can initialize an Agent with knowledge_search_query, like `Agent(knowledge_search_query="Whatever")`, right? We don't added any documentatio about that, but is a public Agent attribute, so... the line 269 will overwrite the pre-defined values. Should we concern about that? is it an expected behavior?